lms/djangoapps/courseware/courses.py

```diff
@@ -142,6 +142,19 @@ def check_course_access(course, user, action, check_if_enrolled=False):
             raise CourseAccessRedirect(reverse('about_course', args=[unicode(course.id)]))
 
 
+def can_user_enroll_in_course(course_key):
```

Comment:
I understand why we wanted a generic name for this, but it almost seems too generic? I feel like it should be merged into something that already exists in `student/models.py` and the logic should live there.